packages/client-sdk-web/src/internal/auth-client.ts

```diff
@@ -164,37 +176,111 @@ function tokenPermissionToGrpcPermission(
 function topicPermissionToGrpcPermission(
   permission: TopicPermission
 ): _GenerateApiTokenRequest.PermissionsType.TopicPermissions {
+  const grpcPermission =
+    new _GenerateApiTokenRequest.PermissionsType.TopicPermissions();
+
   switch (permission.topicRole) {
     case TopicRole.None:
       throw new Error('TopicRole.None not yet supported');
     case TopicRole.ReadWrite: {
-      const grpcPermission =
-        new _GenerateApiTokenRequest.PermissionsType.TopicPermissions();
       grpcPermission.setRole(_GenerateApiTokenRequest.TopicRole.TOPICREADWRITE);
-      return grpcPermission;
+      break;
+    }
+    case TopicRole.ReadOnly: {
+      grpcPermission.setRole(_GenerateApiTokenRequest.TopicRole.TOPICREADONLY);
+      break;
+    }
+    default: {
+      throw new Error(`Unrecognized topic role: ${JSON.stringify(permission)}`);
     }
   }
 
-  throw new Error(
-    `Unrecognized topic permission: ${JSON.stringify(permission)}`
-  );
+  const cacheSelector =
+    new _GenerateApiTokenRequest.PermissionsType.CacheSelector();
+
+  if (permission.cache === AllCaches) {
+    grpcPermission.setAllCaches(
+      new _GenerateApiTokenRequest.PermissionsType.All()
+    );
+  } else if (typeof permission.cache === 'string') {
+    cacheSelector.setCacheName(permission.cache);
+    grpcPermission.setCacheSelector(cacheSelector);
+  } else if (isCacheName(permission.cache)) {
+    cacheSelector.setCacheName(permission.cache.name);
```

Comment:
The `Fine grained authorization scenarios › can read/write cache FGA_CACHE_1 and read/write all topics in cache FGA_CACHE_2` auth integ test is failing because you are missing a `grpcPermission.setCacheSelector(cacheSelector);` here.

Reply:
Creating such a token should not have been allowed. Fixed that in https://github.com/momentohq/control-plane-service/pull/335

Reply:
Thanks, I'll make correction in this PR too